platformio.ini

```diff
@@ -0,0 +1,27 @@
+; PlatformIO Project Configuration File
+;
+;   Build options: build flags, source filter
+;   Upload options: custom upload port, speed and extra flags
+;   Library options: dependencies, extra library storages
+;   Advanced options: extra scripting
+;
+; Please visit documentation for the other options and examples
+; https://docs.platformio.org/page/projectconf.html
+
+[platformio]
+src_dir = OpenBikeSensorFirmware
+
+[env:esp32dev]
+platform = espressif32
+board = esp32dev
+framework = arduino
+# The following dependencies can either use their full name or their (platformio library) id
+lib_deps =
+    # ArduinoJson by Benoit Blanchon
+    ArduinoJson
+    # CircularBuffer by AgileWare
+    CircularBuffer
+    # TinyGPSPlus by Mikal Hart
+    TinyGPSPlus
+    # SSD1306 by ThingPulse
+    562
```

Comment:
```suggestion
    562
; Optional. If not set, PlatformIO will try to autodetect the port
;upload_port = /dev/cu.SLAB_USBtoUART

```

Reply:
Hier ist die Frage ob man die Zeile dazu packen will oder nicht, weil eine lokale Änderung später wieder von Git überschrieben werden kann (wenn jemand Upstream etwas ändert).

Alternativ könnte man auch wie bei unserem Radmesser-Code diese Datei zu `platformio.ini.example` umbennen, sodass ein User diese Datei sich selbst kopieren muss. Das hätte den Vorteil, dass bestimmte Konfigurationen lokal von individuellen Nutzern festgelegt werden können, ohne dass deren Änderungen später von Git gelöscht werden könnten.

Reply:
kann man die irgendwie extenden? Für das beste aus beiden welten?

Reply:
Ich glaube ich habe eine Lösung dafür gefunden, aber konnte die noch nicht testen ...

Reply:
bin interessiert, konnte in den platformio docs keine zufriedenstellende Lösung finden.

Reply:
Habe gerade dazu einen passenden Commit gepushed. Lokal kann man nun eine `custom_config.ini` Datei erstellen die dann die `platformio.ini` Datei erweitert. In der Beispiel `custom_config.ini.example` steht beschrieben wie man manuell den `upload_port` setzen kann. Diese Konfigurationsdatei muss aber nicht erstellt werden und wird auch nicht in git getracked.

Reply:
cool!